test/models/job_execution_test.rb

```diff
@@ -139,8 +141,10 @@
 
   it 'does not export deploy_group information if no deploy groups present' do
     job.update(command: 'env | sort')
-    execute_job
-    job.output.wont_include "DEPLOY_GROUPS"
+    deploy.stub :stage, stage_no_groups do
```

Comment:
Can probably just do a `deploy.stage = stage_no_groups`, without needing to stub. Did you try that?